tests/unit/AccountService_test.js

```diff
@@ -0,0 +1,40 @@
+describe('test AccountService Class',function() {
+  //var songObject = {"image":"../core/assets/2015-best-music.png","songTitle":"love love enemies love","songAuthor":"Smiley","songLength":74,"songListened":82};
+  var accountService;
+  accountService = new AccountService();
+  // beforeEach(function(){
+  //
+  // });
+
+  it('test accountservice get usersArray', function() {
+    expect(accountService.usersArray).toEqual([]);
+  });
+  it('test accountservice set usersname', function() {
+    accountService.setUsername("u");
+    expect(accountService.username).toEqual("u");
+  });
+  it('test accountservice set password', function() {
+    accountService.setPassword("password new");
+    expect(accountService.password).toEqual("password new");
+  });
+  it('test accountservice add user', function() {
+    accountService.addUser({name:"user name"});
+    expect(accountService.usersArray).toEqual([{name:"user name"}]);
+  });
+
+  it('test check login', function() {
+    accountService.setUsername("u");
```

Comment:
add a test also for when only the username is incorrect(and password is correct). You should try to cover all combinations(tip: code coverage can give you good hints about what to test).